mobile/Ihc/services/LocalDataService.js

```diff
@@ -218,30 +218,42 @@ export function getTriage(patientKey, strDate) {
   return triage;
 }
 
-// Update/Create a medication
-export function updateMedication(oldKey, update) {
+// Create a medication
+export function createMedication(medication) {
   const timestamp = new Date().getTime();
-  const medication = realm.objects('Medication')
+  // just incase object passed in does not already have key initialized
+  medication.key = Medication.makeKey(medication);
+  const existingMedication = realm.objects('Medication')
+    .filtered('key = "' + medication.key + '"')['0'];
+
+  medication.lastUpdated = timestamp;
+
+  if (existingMedication) {
+    throw new Error('Medication already exists');
+  }
+
+  realm.write( () => {
+    realm.create('Medication', medication);
+  });
+}
+
+// Update a medication
+export function updateMedication(oldKey, update) {
+  const existingMedication = realm.objects('Medication')
     .filtered('key = "' + oldKey + '"')['0'];
 
-  update.lastUpdated = timestamp;
+  if (!existingMedication) {
+    throw new Error('Medication does not exist');
+  }
 
   realm.write( () => {
-    //Medication already exists (update)
-    if (medication) {
-      const properties = Object.keys(Medication.schema.properties);
-      properties.forEach( p => {
-        medication[p] = update[p];
-      });
-      medication.key = Medication.makeKey(update);
-      return true;
-    }
-    //Medication does not exist (create)
-    update.key = Medication.makeKey(update);
-    realm.create('Medication', update);
-    return true;
+    const properties = Object.keys(Medication.schema.properties);
+    properties.forEach( p => {
+      existingMedication[p] = update[p];
+    });
+    existingMedication.key = Medication.makeKey(update);
```

Comment:
I left a lot of comments in Brent's pull request, so can you also check those out? Specifically, I left a comment about how we shouldn't be changing the key because the point of keys is to be a constant, unique identifier